### DISCUSSION TITLE
Take to result screen after viewing a failed replay

### DISCUSSION THREAD

**Discussion by MicrocontrollersDev** [ppy/osu#30577] [Ideas] [answered]
Currently when watching a failed replay, the game asks to retry or quit once the player fails the map. Quitting takes you to song select and retrying replays the replay. This means its impossible to get to the results screen. This prevents you from viewing the accuracy heatmap or any other statistics that are only available after watching the replay, despite the fact that the results screen tells you that you can watch the replay to see these stats.

The results screen is already viewable from song select and is required to watch the replay anyways, so it makes sense to display that instead. The retry screen also doesn't make sense in the context of a replay, pressing retry just restarts the replay, and the bottom watch replay button has almost the exact same function.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/pull/30617

**Comment by vatei** [comment]
I'd like an option to pause the replay when reaching the end instead of showing the results then, if I'm viewing a failed replay there's a high chance I'm analyzing my mistakes and I wouldn't want the results screen to interrupt that every time.

**Reply by MicrocontrollersDev** [reply]
personally feel this to be true for regular replays as well, not just failed replays. I like playing a lot of shorter maps and when watching regular passed replays I've had this happen on several occasions

**Reply by peppy** [reply]
You can hit `.` or `,` to pause and enter frame skip mode if that helps. Or just space bar.

Arguably we may want a way to have these keys cancel the display of the results screen, such as

```diff
diff --git a/osu.Game/Screens/Play/ReplayPlayer.cs b/osu.Game/Screens/Play/ReplayPlayer.cs
index 5bd775749e..a2abff3988 100644
--- a/osu.Game/Screens/Play/ReplayPlayer.cs
+++ b/osu.Game/Screens/Play/ReplayPlayer.cs
@@ -7,11 +7,13 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using JetBrains.Annotations;
 using osu.Framework.Allocation;
 using osu.Framework.Bindables;
 using osu.Framework.Input.Bindings;
 using osu.Framework.Input.Events;
 using osu.Framework.Screens;
+using osu.Framework.Threading;
 using osu.Game.Beatmaps;
 using osu.Game.Configuration;
 using osu.Game.Input.Bindings;
@@ -33,6 +35,9 @@ public partial class ReplayPlayer : Player, IKeyBindingHandler<GlobalAction>
 
         private readonly bool replayIsFailedScore;
 
+        [CanBeNull]
+        private ScheduledDelegate resultsScreenDisplay;
+
         protected override UserActivity InitialActivity => new UserActivity.WatchingReplay(Score.ScoreInfo);
 
         protected override bool ShowFailOverlay => false;
@@ -128,6 +133,8 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
                     return true;
 
                 case GlobalAction.TogglePauseReplay:
+                    resultsScreenDisplay?.Cancel();
+
                     if (GameplayClockContainer.IsPaused.Value)
                         GameplayClockContainer.Start();
                     else
@@ -140,6 +147,8 @@ public bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
 
         public void StepFrame(int direction)
         {
+            resultsScreenDisplay?.Cancel();
+
             GameplayClockContainer.Stop();
 
             var frames = GameplayState.Score.Replay.Frames;
@@ -155,6 +164,8 @@ public void StepFrame(int direction)
 
         public void SeekInDirection(float amount)
         {
+            resultsScreenDisplay?.Cancel();
+
             double target = Math.Clamp(GameplayClockContainer.CurrentTime + amount * BASE_SEEK_AMOUNT, 0, GameplayState.Beatmap.GetLastObjectTime());
 
             Seek(target);
@@ -163,7 +174,7 @@ public void SeekInDirection(float amount)
         protected override void OnFail()
         {
             // Replays will always show the results screen on failing.
-            Scheduler.AddDelayed(() =>
+            resultsScreenDisplay = Scheduler.AddDelayed(() =>
             {
                 if (!this.IsCurrentScreen())
                     // This player instance may already be in the process of exiting.

```

(this needs extra attention to reverse the fail animation though.

**Reply by vatei** [reply]
> You can hit . or , to pause and enter frame skip mode if that helps. Or just space bar.

That didn't seem to do anything (using azerty layout though)
Automatically pausing the replay at the end removes the guessing, I think it's more convenient imo

**Reply by peppy** [reply]
Currently you'll have to hit the keys before you reach the point of fail.
